stdlib/public/core/StringObject.swift

```diff
@@ -96,13 +100,45 @@ extension _StringObject {
   @inline(__always)
   // TODO: private
   internal
-  init(rawBits: _RawBitPattern) {
+  init(taggedRawBits: _RawBitPattern) {
+#if arch(i386) || arch(arm)
+    self.init(Builtin.reinterpretCast(taggedRawBits), taggedRawBits &>> 32)
```

Comment:
Hm, interesting. I would've expected both of these parameter expressions to blow up. (Due to UInt64 and _Variant not having the same size in reinterpretCast, and the UInt64/UInt type mismatch with &>>. Clearly I’m missing something obvious, but what?